### ISSUE TITLE
Support for SnoopCompile v3

### ISSUE THREAD

**Issue by kimikage** [julia-actions/julia-invalidations#18]
The development branch of SnoopCompile has moved to a new major version v3.
https://github.com/timholy/SnoopCompile.jl/pull/384
Proposals: https://github.com/timholy/SnoopCompile.jl/issues/382

~This action does not appear to cap on the version upper bounds for dependent packages.
Of course, that can be addressed by the user, but definitely should be addressed in this action.~ **Edit:** cf. PR #22

**Comment by schillic** [comment]
SnoopCompile v3 was just released and now this action crashes:

```julia
ERROR: LoadError: UndefVarError: `@snoopr` not defined
```

**Comment by schillic** [comment]
SnoopCompile v2 is not compatible with Julia v1.11:

```julia
ERROR: LoadError: MethodError: no method matching getindex(::Core.MethodInstance, ::Int64)SnoopCompile
```

Since Julia v1.11 was released, fixing to v2 (#22) causes problems.
Is it planned to support SnoopCompile v3 or is there another recommendation (e.g., to restrict the Julia version to v1.10)?
